tests-blackbox/routes/server/schema.test.ts

```diff
@@ -0,0 +1,560 @@
+				// Action
+				const response = await request(getUrl(vendor))
+					.get('/server/schema/snapshot')
+					.set('Authorization', `Bearer ${common.USER.APP_ACCESS.TOKEN}`);
+				const response2 = await request(getUrl(vendor))
+					.get('/server/schema/snapshot')
+					.set('Authorization', `Bearer ${common.USER.API_ONLY.TOKEN}`);
+				const response3 = await request(getUrl(vendor))
+					.get('/server/schema/snapshot')
+					.set('Authorization', `Bearer ${common.USER.NO_ROLE.TOKEN}`);
+
+				// Assert
+				expect(response.statusCode).toEqual(403);
+				expect(response2.statusCode).toEqual(403);
+				expect(response3.statusCode).toEqual(403);
+			});
+		});
+
+		describe('retrieves a snapshot (JSON)', () => {
+			it.each(vendors)(
+				'%s',
+				async (vendor) => {
+					// Action
+					const response = await request(getUrl(vendor))
+						.get('/server/schema/snapshot')
+						.set('Authorization', `Bearer ${common.USER.ADMIN.TOKEN}`);
+
+					// Assert
+					expect(response.statusCode).toEqual(200);
+
+					snapshotsCacheOriginal[vendor] = response.body.data;
+				},
+				300000
+			);
+		});
+
+		describe('retrieves a snapshot (YAML)', () => {
+			it.each(vendors)(
+				'%s',
+				async (vendor) => {
+					// Action
+					const response = await request(getUrl(vendor))
+						.get('/server/schema/snapshot')
+						.query({ export: 'yaml' })
+						.set('Authorization', `Bearer ${common.USER.ADMIN.TOKEN}`);
+
+					// Assert
+					expect(response.statusCode).toEqual(200);
+
+					snapshotsCacheOriginalYaml[vendor] = response.text;
+				},
+				300000
+			);
+		});
+
+		describe('remove tables', () => {
+			describe.each(common.PRIMARY_KEY_TYPES)('%s primary keys', (pkType) => {
+				it.each(vendors)(
+					'%s',
+					async (vendor) => {
+						for (const setDefaultValues of [false, true]) {
+							// Delete existing collections
+							await deleteAllCollections(vendor, pkType, setDefaultValues);
+						}
+
+						await assertCollectionsDeleted(vendor, pkType);
+					},
+					300000
+				);
+			});
+		});
+	});
+
+	common.ClearCaches();
+
+	describe('retrieves empty snapshot', () => {
+		it.each(vendors)(
+			'%s',
+			async (vendor) => {
+				// Action
+				const response = await request(getUrl(vendor))
+					.get('/server/schema/snapshot')
+					.set('Authorization', `Bearer ${common.USER.ADMIN.TOKEN}`);
+
+				// Assert
+				expect(response.statusCode).toEqual(200);
+				expect(snapshotsCacheEmpty[vendor]).not.toEqual(snapshotsCacheOriginal[vendor]);
+
+				snapshotsCacheEmpty[vendor] = response.body.data;
+			},
+			300000
+		);
+	});
+
+	describe('POST /server/schema/diff', () => {
+		describe('denies non-admin users', () => {
+			it.each(vendors)('%s', async (vendor) => {
+				// Action
+				const response = await request(getUrl(vendor))
+					.post('/server/schema/diff')
+					.send({
+						version: 1,
+						directus: currentDirectusVersion,
+						// TODO: align vendor names
+						vendor: 'postgres',
```

Comment:
(commenting here only to point out the TODO)

Reply:
@licitdev I assume the aligning here means between the blackbox test's vendor list:

https://github.com/directus/directus/blob/d2b5881a1211fdce9091495756f3ff2e1f571ba6/tests-blackbox/common/get-dbs-to-test.ts#L1-L11

against the current list in the API:

https://github.com/directus/directus/blob/d2b5881a1211fdce9091495756f3ff2e1f571ba6/api/src/types/database.ts#L3

because we added validation based on the API's list:

https://github.com/directus/directus/blob/d2b5881a1211fdce9091495756f3ff2e1f571ba6/api/src/services/schema.ts#L18-L20

so `postgres10`, `mysql5` and `sqlite3` would indeed be caught by the valdation (I do recall we had the sqlite vs sqlite3 discussion on discord). Any thoughts on how do we want to address this TODO?

Based on the current values, I did thought about maybe we can simply strip the numbers (eg. `postgres10` would become `postgres`) to satisfy the blackbox test, such as:

```diff
					.send({
						version: 1,
						directus: currentDirectusVersion,
-						// TODO: align vendor names
-						vendor: 'postgres',
+						vendor: vendor.replace(/[0-9]/g, ''),
						collections: [],
						fields: [],
						relations: [],
					})
```

but not sure if that is too naive/hacky of a solution 😄

Reply:
@azrikahar Your solution should work well for now, until there's custom vendors / drivers. 💯  
cc @jaads